contrib/runners/orquesta_runner/orquesta_runner/orquesta_runner.py

```diff
@@ -198,8 +199,20 @@ def task_cancelable(ac_ex):
         return wf_ex_cancelable or ac_ex_cancelable
 
     def cancel(self):
-        # Cancel the target workflow.
-        wf_svc.request_cancellation(self.execution)
+        # Try to cancel the target workflow execution.
+        try:
+            wf_svc.request_cancellation(self.execution)
+        # If workflow execution is not found because the action execution is cancelled
+        # before the workflow execution is created or if the workflow execution is
+        # already completed, then ignore the exception and proceed with cancellation.
+        except (wf_svc_exc.WorkflowExecutionNotFoundException,
+                wf_svc_exc.WorkflowExecutionIsCompletedException):
+            pass
+        # If there is an unknown exception, then log and rethrow the exception.
+        except Exception as e:
```

Comment:
Not sure if we can add test case for  `Unable to cancel workflow execution`?

Reply:
@jinpingh Good catch. I changed how the unknown exception is handled and added unit test.